src/cache-cleaner.ts

```diff
@@ -42,9 +44,15 @@ export class CacheCleaner {
         )
         fs.writeFileSync(path.resolve(cleanupProjectDir, 'build.gradle'), 'task("noop") {}')
 
-        await exec.exec(`gradle -g ${this.gradleUserHome} --no-daemon --build-cache --no-scan --quiet noop`, [], {
-            cwd: cleanupProjectDir
-        })
+        const gradleExecutable = this.gradleVersion === 'wrapper' ? './gradlew' : 'gradle'
```

Comment:
could be it can be combined with `gradle-executable` as described here (to handle cases where people can have `gradlew` not in the root folder)
<img width="877" alt="Screenshot 2022-09-12 at 14 37 50" src="https://user-images.githubusercontent.com/4134993/189644020-1e763a1c-bd73-4862-857b-7f742573df44.png">

Reply:
Very true! I've added support for an explicitly passed Gradle executable 😄